### PR TITLE
p2p: Include (hard-coded) frame data in msg headers

### PR DESCRIPTION
**Pull request by gsalgado**
Closes: #718
Closes: #819

### CODE REVIEW

**Comment by veox** [comment]
I'm testing this locally, and it seems that the hadshake still fails.

I've modified `DEFAULT_PREFERRED_NODES` to have:

``` python
    ROPSTEN_NETWORK_ID: (
        Node(keys.PublicKey(decode_hex("053d2f57829e5785d10697fa6c5333e4d98cc564dbadd87805fd4fedeb09cbcb642306e3a73bd4191b27f821fb442fcf964317d6a520b29651e7dd09d1beb0ec")),  # noqa: E501                                           
             Address("79.98.29.154", 30303, 30303)),
        Node(keys.PublicKey(decode_hex("3e780f1fcccbab817bb74fdfb026f2c8fca53a610f42a488eafee1c415aa263a66bdf4424dc40fbbb6a11fb8c134c8c5eb60877087ca5c642da9c995876f6df0")),  # noqa: E501                                           
             Address("207.148.76.54", 30303, 30303)),
```

(The first is already in the `py-evm` repo, is a `geth` node with `lightserv` enabled; the second is a `parity` node - not worth adding to the list permanently yet.)

**Comment by gsalgado** [comment]
> 2018-06-08 11:42:18,073 DEBUG: Successfully decoded Hello (cmd_id=0) msg: {'version': 5, 'client_version_string': 'Parity/v1.12.0-unstable-1318f536c-20180606/x86_64-linux-gnu/rustc1.26.2', 'capabilities': (('eth', 62), ('eth', 63), ('par', 1), ('par', 2), ('par', 3), ('pip', 1)), 'listen_port': 30303, 'remote_pubkey': b'+\x8dZ\xd2\xd9f\x07\xd5\xb9\xe6n\xa9;\xcf&\xe1\x06\xf8P-\x9d\xab\x85Z\xaa1\xd9K\x8a\x9f\x17\xfe\xc2e\x9d\xbc\xfb\x8bu+d\x11T6\x8f\x07\x9d\xcf\x85\xe4\x02\xeaV\x99\xcf\xd2\x05\x13d\x17\xa0m\xc4\xe2'}
> ...
> 2018-06-08 11:42:18,096 DEBUG: Successfully decoded Status (cmd_id=16) msg: {'protocol_version': 63, 'network_id': 3, 'td': 2191941708522, 'best_hash': b'z\x81\xd6\xb8\x93\x11\xf7\x01z\x1f\xac\x01\xbf\xfa\xe2\x96\x92)\xf0x.\x81\xea2\xfa\x02b\x89\x0bM\x10\xcd', 'genesis_hash': b'A\x94\x10#h\t#\xe0\xfeMt\xa3K\xda\xc8\x14\x1f%@\xe3\xae\x90b7\x18\xe4}f\xd1\xcaJ-'}
2018-06-08 11:42:18,100 DEBUG: Finished (eth, 63) handshake with <Node(0x2b8d@127.0.0.1)>
2018-06-08 11:42:18,101 INFO: Successfully connected to ETHPeer <Node(0x2b8d@127.0.0.1)>

Locally it works for me against parity 1.7.2 and 1.12.0

File: tests/p2p/integration_test_helpers.py [hunk 1]
```diff
@@ -37,11 +37,13 @@ def __init__(self,
         )
 
     def get_nodes_to_connect(self):
-        nodekey = keys.PrivateKey(decode_hex(
-            "45a915e4d060149eb4365960e6a7a45f334393093061116b197e3240065ff2d8"))
-        remoteid = nodekey.public_key.to_hex()
-        yield kademlia.Node(keys.PublicKey(decode_hex(remoteid)),
+        # local-geth's pubkey
+        # pubkey_hex = "3a514176466fa815ed481ffad09110a2d344f6c9b78c1d14afc351c3a51be33d8072e77939dc03ba44790779b7a1025baf3003f6732430e20cd9b76d953391b3"  # noqa: E501
```
Comment (carver) [inline tests/p2p/integration_test_helpers.py]: Is this a local change that slipped into the PR?

Comment (gsalgado) [reply tests/p2p/integration_test_helpers.py]: No, this class (`LocalGethPeerPool`) is a bit of a hack to help me manually test things against a local node. Before it had only the node ID of my geth instance, I just added the node ID for my parity instance now

**Comment by gsalgado** [comment]
@carver are you :+1: on this, then?

**Pull request merged**